Source/Core/Core/Boot/Boot.cpp

```diff
@@ -38,6 +41,57 @@
 #include "DiscIO/NANDContentLoader.h"
 #include "DiscIO/Volume.h"
 
+BootParameters::BootParameters(Parameters&& parameters_) : parameters(std::move(parameters_))
+{
+}
+
+std::optional<BootParameters> BootParameters::GenerateFromFile(const std::string& path)
+{
+  const bool is_drive = cdio_is_cdrom(path);
+  // Check if the file exist, we may have gotten it from a --elf command line
+  // that gave an incorrect file name
+  if (!is_drive && !File::Exists(path))
+  {
+    PanicAlertT("The specified file \"%s\" does not exist", path.c_str());
+    return {};
+  }
+
+  std::string extension;
+  SplitPath(path, nullptr, nullptr, &extension);
+  std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
+
+  const std::unordered_set<std::string> disc_image_extensions = {
+      {".gcm", ".iso", ".tgc", ".wbfs", ".ciso", ".gcz"}};
```

Comment:
This list (and the `IsDiscFile` check, really) feels duplicated from somewhere else; like it belongs to a different shared location where everyone can call it...but I don't really remember any other location where they're used like that.
Can just leave it here if this is the only place though.